site/src/contexts/usePushNotifications.ts

```diff
@@ -0,0 +1,105 @@
+import { buildInfo } from "api/queries/buildInfo";
+import { useQuery } from "react-query";
+import { useEmbeddedMetadata } from "hooks/useEmbeddedMetadata";
+import { useEffect, useState } from "react";
+import { API } from "api/api";
+
+interface PushNotifications {
+    readonly subscribed: boolean;
+    readonly loading: boolean;
+
+    subscribe(): Promise<void>;
+    unsubscribe(): Promise<void>;
+}
+
+export const usePushNotifications = (): PushNotifications => {
+    const { metadata } = useEmbeddedMetadata();
+	const buildInfoQuery = useQuery(buildInfo(metadata["build-info"]));
+	
+    const [subscribed, setSubscribed] = useState<boolean>(false);
+    const [loading, setLoading] = useState<boolean>(true);
+
+    useEffect(() => {
+        // Check if browser supports push notifications
+        if (!("Notification" in window) || !("serviceWorker" in navigator)) {
+            setSubscribed(false);
+            setLoading(false);
+            return;
+        }
+
+        const checkSubscription = async () => {
+            try {
+                const registration = await navigator.serviceWorker.ready;
+                const subscription = await registration.pushManager.getSubscription();
+                setSubscribed(!!subscription);
+            } catch (error) {
+                console.error("Error checking push subscription:", error);
+                setSubscribed(false);
+            } finally {
+                setLoading(false);
+            }
+        };
+
+        checkSubscription();
+    }, []);
+
+    const subscribe = async (): Promise<void> => {
+        try {
+            setLoading(true);
+            const registration = await navigator.serviceWorker.ready;
+
+            console.log("BUILD INFO", buildInfoQuery.data?.push_notifications_public_key);
+            
+            // Note: You'd typically get this key from your server
+            const vapidPublicKey = buildInfoQuery.data?.push_notifications_public_key;
```

Comment:
From reading [RFC8292](https://datatracker.ietf.org/doc/rfc8292/), section 4.2 states:

```
   A push service MUST NOT forward the JWT or public key to the user
   agent when delivering the push message.
```

Does this count?